optimum/onnxruntime/modeling_ort.py

```diff
@@ -250,6 +261,14 @@ def device(self) -> torch.device:
     def device(self, value: torch.device):
         self._device = value
 
+    @property
+    def use_io_binding(self):
+        return check_io_binding(self.providers, self._use_io_binding)
+
+    @use_io_binding.setter
+    def use_io_binding(self, value: bool):
```

Comment:
This is actually very important, because the way it was done previously did not handle this case:

```python
my_model = ORTModel(..., providers=["CPUExecutionProvider"]).to("cuda")
```

Before this, we would not use IO binding because `self.use_io_binding` was set at initialization.